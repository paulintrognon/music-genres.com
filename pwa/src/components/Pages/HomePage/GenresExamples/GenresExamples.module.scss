.container {
  margin-top: 1.5rem;
  text-align: center;
  font-size: var(--font-size-14);
}

.example {
  display: inline-block;
  background-color: var(--color-secondary);
  box-shadow: 0 9px 10px 0 rgba(0,0,0,.23);
  padding: 0.35rem 0.75rem;
  margin-left: 1rem;
  margin-bottom: 0.5rem;
  transition: background-color 200ms;

  &:hover {
    background-color: var(--color-secondary-light);
  }
}
