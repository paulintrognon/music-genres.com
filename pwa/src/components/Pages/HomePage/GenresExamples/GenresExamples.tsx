import Link from 'next/link'
import useSWR from 'swr'
import { ExampleGenresApiResponseType } from '../../../../types/Api/ExampleGenresApiResponseType'
import Loader from '../../../Atoms/Loader/Loader'
import classes from './GenresExamples.module.scss'

const GenresExamples: React.FC = () => {
  /**
   * On query state changes, we fetch 3 genres examples
   */
  const { isValidating, data } = useSWR<ExampleGenresApiResponseType>('/api/music-genres/random')
  if (!data) {
    return null
  }

  return (
    <div className={classes.container}>
      Examples:
      {isValidating ? (
        <>
          <a className={classes.example}>
            <Loader />
          </a>
          <a className={classes.example}>
            <Loader />
          </a>
          <a className={classes.example}>
            <Loader />
          </a>
        </>
      ) : (
        data.result.map((genre) => (
          <Link key={genre.slug} href={`/genre/${genre.slug}`}>
            <a className={classes.example}>{genre.name}</a>
          </Link>
        ))
      )}
    </div>
  )
}
export default GenresExamples
