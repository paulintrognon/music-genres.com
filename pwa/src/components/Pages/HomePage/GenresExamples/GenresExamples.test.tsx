import GenresExamples from './GenresExamples'

// Mocking getPost
import useSWR from 'swr'
import { shallow } from 'enzyme'
jest.mock('swr')
const useSWRMock = useSWR as jest.Mock

beforeEach(() => {
  jest.resetAllMocks()
})

it('should render examples after fetch', async () => {
  const examplesResult = [
    { name: 'Genre 1', slug: 'genre-1' },
    { name: 'Genre 2', slug: 'genre-2' },
    { name: 'Genre 3', slug: 'genre-3' },
  ]
  useSWRMock.mockReturnValueOnce({ data: { result: examplesResult } })
  const wrapper = shallow(<GenresExamples />)
  expect(wrapper).toMatchSnapshot()

  expect(useSWRMock).toHaveBeenCalledTimes(1)
  expect(useSWRMock.mock.calls[0]).toEqual(['/api/music-genres/random'])
})

it('should render with loaders if loading', async () => {
  useSWRMock.mockReturnValueOnce({ isValidating: true, data: {} })
  const wrapper = shallow(<GenresExamples />)
  expect(wrapper).toMatchSnapshot()
})

it('should return null if no data', async () => {
  useSWRMock.mockReturnValueOnce({ data: null })
  const wrapper = shallow(<GenresExamples />)
  expect(wrapper).toMatchSnapshot()
})
