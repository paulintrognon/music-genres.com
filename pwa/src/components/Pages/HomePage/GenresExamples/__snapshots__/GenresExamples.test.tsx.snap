// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render examples after fetch 1`] = `
<div
  className="container"
>
  Examples:
  <Link
    href="/genre/genre-1"
    key="genre-1"
  >
    <a
      className="example"
    >
      Genre 1
    </a>
  </Link>
  <Link
    href="/genre/genre-2"
    key="genre-2"
  >
    <a
      className="example"
    >
      Genre 2
    </a>
  </Link>
  <Link
    href="/genre/genre-3"
    key="genre-3"
  >
    <a
      className="example"
    >
      Genre 3
    </a>
  </Link>
</div>
`;

exports[`should render with loaders if loading 1`] = `
<div
  className="container"
>
  Examples:
  <a
    className="example"
  >
    <Loader />
  </a>
  <a
    className="example"
  >
    <Loader />
  </a>
  <a
    className="example"
  >
    <Loader />
  </a>
</div>
`;

exports[`should return null if no data 1`] = `""`;
